CodingConventions.md

```diff
@@ -0,0 +1,70 @@
+# Code conventions for AIrsenal
+
+This note aims to capture some of the practices we have (consciously or
+unconsciously) adopted during the development of AIrsenal, with a view
+to making the code more consistent, and therefore easier to develop.
+
+It is not intended as a set of hard-and-fast rules - there will always be
+exceptions, and we definitely don't want to deter anyone from contributing,
+rather we hope that this will develop into a set of helpful guidelines, and
+additions/corrections to this document are always welcome!
+
+## Git branches
+
+In the original AIrsenal repo in the "alan-turing-institute" organization, we
+maintain two long-running branches: "main" and "develop".
+The aim is that "main" will always be functional, and this is likely to be the
+first entrypoint to the project for new users.
+The "develop" branch should hopefully also be functional, but it is not inconceivable that merging a new feature could break it for brief periods.
+
+Development should mostly take place on individual branches that are branched off "main" or "develop".   As a general guideline, new features should be branched off, and merged back into "develop", while bugfixes can be branched off, and merged back into "main" (and then merge "main" into "develop").
+The naming convention for branches should generally be something like
+`feature/<github_issue_number>-<brief-description>` or `bugfix/<github_issue_number>-<brief-description>` (and as this implies, there should ideally be a corresponding Issue!).
+
+## Code style, formatting, code quality
+
+We are generally following the [PEP-8 style guide][link_pep8] regarding conventions for class, function, and variable names.
+
+Ideally, docstrings should follow [numpydoc][link_numpydoc] convention (though this is not always the case in the existing code).
+We encourage extensive documentation.
+
```

Comment:
```suggestion

Although there are not many in the current codebase, we also encourage the use of type hints, as provided by the [typing](link_typing) module. Examples of functions using this can be found in `airsenal/framework/player_model.py`.
```